ARC based boards don't build with SDK 0.9.5

We updated binutils to 2.29 for arc after SDK 0.9.3, this seems to cause ARC platforms not to build due to:

```
FAILED: zephyr/zephyr_prebuilt.elf 
: && ccache /opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/arc-zephyr-elf-gcc    zephyr/CMakeFiles/zephyr_prebuilt.dir/misc/empty_file.c.obj  -o zephyr/zephyr_prebuilt.elf  -T zephyr/linker.cmd -Wl,-Map=/home/galak/git/zephyr/sanity-out/quark_se_c1000_ss_devboard/samples/hello_world/sample.helloworld/zephyr/zephyr.map -u_OffsetAbsSyms -u_ConfigAbsSyms -Wl,--whole-archive libapp.a zephyr/libzephyr.a zephyr/arch/arc/core/libarch__arc__core.a zephyr/lib/libc/minimal/liblib__libc__minimal.a zephyr/ext/hal/qmsi/libext__hal__qmsi.a zephyr/drivers/serial/libdrivers__serial.a -Wl,--no-whole-archive zephyr/kernel/libkernel.a zephyr/CMakeFiles/offsets.dir/arch/arc/core/offsets/offsets.c.obj -L"/opt/zephyr-sdk-0.9.5/sysroots/arc-zephyr-elf/usr/lib/arc-zephyr-elf/6.2.1/quarkse_em" -L/home/galak/git/zephyr/sanity-out/quark_se_c1000_ss_devboard/samples/hello_world/sample.helloworld/zephyr -lgcc -Wl,--print-memory-usage -nostdlib -static -no-pie -Wl,--fatal-warnings -Wl,-X -Wl,-N -Wl,--gc-sections -Wl,--build-id=none -Wl,--orphan-handling=warn && :
Memory region         Used Size  Region Size  %age Used
           FLASH:        7872 B       152 KB      5.06%
            SRAM:        3732 B        24 KB     15.19%
            DCCM:          0 GB         8 KB      0.00%
        IDT_LIST:          25 B         2 KB      1.22/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `libapp.a(main.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/libzephyr.a(isr_tables.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/libzephyr.a(thread_entry.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/libzephyr.a(printk.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/libzephyr.a(configs.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/libzephyr.a(soc.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/libzephyr.a(uart_console.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/libzephyr.a(arcv2_irq_unit.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/libzephyr.a(sys_clock_init.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/libzephyr.a(arcv2_timer0.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(thread.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(thread_entry_wrapper.S.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(cpu_idle.S.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(fatal.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(fault.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(fault_s.S.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(irq_manage.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(isr_wrapper.S.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(regular_irq.S.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(swap.S.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(sys_fatal_error_handler.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(prep_c.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(reset.S.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(vector_table.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/arch/arc/core/libarch__arc__core.a(fast_irq.S.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/lib/libc/minimal/liblib__libc__minimal.a(string.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/ext/hal/qmsi/libext__hal__qmsi.a(clk.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/ext/hal/qmsi/libext__hal__qmsi.a(qm_uart.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/drivers/serial/libdrivers__serial.a(uart_qmsi.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/kernel/libkernel.a(device.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/kernel/libkernel.a(errno.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/kernel/libkernel.a(init.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/kernel/libkernel.a(sched.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/kernel/libkernel.a(thread.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/kernel/libkernel.a(thread_abort.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/kernel/libkernel.a(timeout.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/kernel/libkernel.a(idle.c.obj)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `zephyr/CMakeFiles/offsets.dir/arch/arc/core/offsets/offsets.c.obj' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `/opt/zephyr-sdk-0.9.5/sysroots/arc-zephyr-elf/usr/lib/arc-zephyr-elf/6.2.1/quarkse_em/libgcc.a(_millicodethunk_st.o)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `/opt/zephyr-sdk-0.9.5/sysroots/arc-zephyr-elf/usr/lib/arc-zephyr-elf/6.2.1/quarkse_em/libgcc.a(_millicodethunk_ld.o)' being placed in section `.ARC.attributes'.
/opt/zephyr-sdk-0.9.5/sysroots/x86_64-pokysdk-linux/usr/bin/arc-zephyr-elf/../../libexec/arc-zephyr-elf/gcc/arc-zephyr-elf/6.2.1/real-ld: warning: orphan section `.ARC.attributes' from `/opt/zephyr-sdk-0.9.5/sysroots/arc-zephyr-elf/usr/lib/arc-zephyr-elf/6.2.1/quarkse_em/libgcc.a(_millicodethunk_ret.o)' being placed in section `.ARC.attributes'.
collect2: error: ld returned 1 exit status
%
```

--- a/include/arch/arc/v2/linker.ld
+++ b/include/arch/arc/v2/linker.ld
@@ -289,4 +289,11 @@ SECTIONS {
 
 #include <linker/debug-sections.ld>
 
+
+    SECTION_PROLOGUE(.ARC.attributes, 0,)
+	{
+	KEEP(*(.ARC.attributes))
+	KEEP(*(.gnu.attributes))
+	}
+
 	}

